modules/deployment-scripts/create-agw-kv-certificate/main.bicep

```diff
@@ -0,0 +1,229 @@
+  'a4417e6f-fecd-4de8-b567-7b0420556985' //KeyVault Certificate Officer
+]
+
+@description('An array of Azure Application Gateway RoleIds that are required for the DeploymentScript resource')
+param rbacRolesNeededOnAppGw array = [
+  'b24988ac-6180-42a0-ab88-20f7382dd24c' //Contributor
+]
+
+@description('Does the Managed Identity already exists, or should be created')
+param useExistingManagedIdentity bool = false
+
+@description('Name of the Managed Identity resource')
+param managedIdentityName string = 'id-KeyVaultCertificateCreator'
+
+@description('For an existing Managed Identity, the Subscription Id it is located in')
+param existingManagedIdentitySubId string = subscription().subscriptionId
+
+@description('For an existing Managed Identity, the Resource Group it is located in')
+param existingManagedIdentityResourceGroupName string = resourceGroup().name
+
+@description('An array of Certificate names to create')
+param certNames array
+
+@allowed([
+  'none'
+  'root-cert'
+  'ssl-cert'
+])
+@description('Configured certificate in Application Gateway as Frontend (ssl-cert) or Backend (root-cert)')
+param agwCertType string = 'none'
+
+@description('A delay before the script import operation starts. Primarily to allow Azure AAD Role Assignments to propagate')
+param initialScriptDelay string = '30s'
+
+@description('A delay before the Application Gateway imports the Certificate from KeyVault. Primarily to allow the certificate creation operation to complete')
+param agwCertImportDelay string = '120s'
+
+@allowed([
+  'OnSuccess'
+  'OnExpiration'
+  'Always'
+])
+@description('When the script resource is cleaned up')
+param cleanupPreference string = 'OnSuccess'
+
+resource akv 'Microsoft.KeyVault/vaults@2021-11-01-preview' existing = {
+  name: akvName
+}
+
+resource agw 'Microsoft.Network/applicationGateways@2021-05-01' existing = if (!empty(agwName) && agwCertType != 'none') {
+  name: agwName
+}
+
+@description('The User Assigned Identity of the Azure Application Gateway')
+param agwIdName string = ''
+resource agwId 'Microsoft.ManagedIdentity/userAssignedIdentities@2018-11-30' existing = if (!empty(agwName) && !empty(agwIdName) && agwCertType != 'none') {
+  name: agwIdName
+}
+
+@description('A new managed identity that will be created in this Resource Group, this is the default option')
+resource newDepScriptId 'Microsoft.ManagedIdentity/userAssignedIdentities@2018-11-30' = if (!useExistingManagedIdentity) {
+  name: managedIdentityName
+  location: location
+}
+
+@description('An existing managed identity that could exist in another sub/rg')
+resource existingDepScriptId 'Microsoft.ManagedIdentity/userAssignedIdentities@2018-11-30' existing = if (useExistingManagedIdentity ) {
+  name: managedIdentityName
+  scope: resourceGroup(existingManagedIdentitySubId, existingManagedIdentityResourceGroupName)
+}
+
+resource rbacKv 'Microsoft.Authorization/roleAssignments@2020-08-01-preview' = [for roleDefId in rbacRolesNeededOnKV: {
```

Comment:
unsupported scenario. I'll make this clear in the readme

Reply:
I think RBAC only is fine, we should just make it clear...  also wondering if we need to do something beyond the readme (full disclosure: no idea what that would be...)

Reply:
Not sure there's a good answer here.

We could elevate the Identity permission to be able to check if the AKV is using RBAC instead of access policies in the script... Definitely increasing the fault surface area though.

The *best* way would be to reference the akv.properties.enableRbacAuthorization as a resource condition. However it needs to be calculated at the start.

Suggest that we just document in the readme for the time being. I'll leave this discussion open until we have a conclusion